packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -16,9 +16,18 @@ import { InvocationCallResult, FunctionInvocationResult } from './invocation-res
 function createContractCall(funcScope: InvocationScopeLike): ContractCall {
   const { program, args, forward, func, callParameters, bytesOffset } = funcScope.getCallConfig();
 
+  const provider = program.provider as Provider;
+  const consensusParams = provider.consensusParamsCache;
+
+  if (!consensusParams) {
+    throw new Error(
+      'Provider consensus params cache empty! Pls make sure you ran `await Provider.connect()` and not just `new Provider()`'
+    );
+  }
```

Comment:
This feels a bit unintuitive to me, we should probably ensure that `Provider::connect()` is called within the `constructor()` and thus has the consensus params available otherwise I can see us needing this check in multiple places. And then we can also ensure type safety that `consensusParams` is always set on the `Provider` instance.